### ISSUE TITLE
The state of folds isn't saved by mkview

### ISSUE THREAD

**Issue by Shamaoke** [vim/vim#6274]
Opened/closed state of folds isn't saved by the `mkview` command. The view is always restored with folds in the closed state.

1. `touch sample.txt`;

2. `vim --clean sample.txt`

3. Type any text and make some folds by hand (using `zf`), e. g.

```
|do
|
| first fold
|
|end

|do
|
| second fold
|
|end

|do
|
| third fold
|
|end
```

4. Write the changes;

5. Open all folds by `zR`;

6. Create a view by `mkview` with folds opened;

7. Load the view by `loadview` — the view will be loaded with all folds closed.

_Linux (Arch); Vim 8.2; urxvt 9.22_

**Comment by liskin** [comment]
I'm observing this as well. It seems `mkview`, after `zR`, doesn't save any `zo`/`zc` at all. For other foldmethods, this is taken care of by setting `fdl` after `fdm` and `fde` and `fmr`, but with `foldmethod=manual`, the folds are created after setting `fdl`, so they're not opened.

**Comment by chrisbra** [comment]
Isn't ther problem here rather, that there does not exists a way to save and restore manual folds at all? For all other fold methods, this is taken care by the rules implied on the fold method. But for manual folds, we simply do not know.

**Comment by liskin** [comment]
No, that's not it at all, the folds are saved all right.

In the view script, there's a bunch of `setlocal`s that set `fdm` and `fdl` and `fen` and others, then there's a bunch of `{range}fold` commands to create the manual folds (these aren't there for the automatic fold methods), and then there'd normally be a bunch of jumps and `zo`/`zc`s, but these aren't there after `zR`, and that's a problem, because `fold` creates the fold closed:

```
zf{motion}  or
{Visual}zf	Operator to create a fold.
		This only works when 'foldmethod' is "manual" or "marker".
		The new fold will be closed for the "manual" method.
		'foldenable' will be set.
		Also see |fold-create-marker|.

:{range}fo[ld]						*:fold* *:fo*
		Create a fold for the lines in {range}.  Works like "zf".
```

**Comment by liskin** [comment]
I think we can close this now, #7767 fixes this and I've verified that I can no longer reproduce it with current master vim.

**Comment by brammool** [comment]
Fixed by patch 8.2.2447
